### PR TITLE
Addition of user details to review form

### LINKED ISSUE #1137: NPS: Add more data to user review [/1]
When we submit a user review ("Would you recommend Mr Beam to your friends? 1-10?") we need to add more device data and also ask the user more details.

Device data:
- SW version (mr beam plugin)
- model (MRBEAM2_DC, ...)
- production date (be aware that older devices do not have a model_id or production date)

Server side:
We need to add the date when the review was uploaded (which might be very different from the date when the review was done.) The creation date of the json file in the GCP bucket could be used. All following components need to user or forward this date.

User data:
- We should ask the user if it's ok to contact her/him "in case we have questions" and offer to add email address or phone number. Could be one (free text) field or two fields. We could also pre-fill the user's email address, but I lean towards not doing this. We probably also need a statement what we use these data for. 
- the user should have the option to change it's rating. Since we asks in any case for feedback and contact details, it does not change the logic/flow of the user review screen.
- please remove the "wow thank you" text as if we were surprised about a good rating. Let's have a neutral "Thank you for your feedback" or "Your feedback is super important to us" or so.
- (Optional) If the user gave us a good rating, we could add another screen AFTER he submitted his complete feedback. This new screen would be dedicated to suggest that the suer rates us on rating pages. this page should get a visual makeover I think. In my mind we should show there logos of well known rating platforms. But I need to check with Jessie. I see this screen as optional, we can add it later on as well.
  - Our Shop (as it is currently, but please let's use a redirect `mr-beam.org/beamos/review`) 
  - Google
  - Trustpilot

### PR DESCRIPTION
**Pull request by khaledsherkawi**

### CODE REVIEW

**Review by andywerner — CHANGES_REQUESTED** [review]
Please change as we discussed today on the phone.
In the issue I wrote the wording : https://github.com/mrbeam/MrBeamPlugin/issues/1137#issuecomment-776919717

**Comment by khaledsherkawi** [comment]
The only thing I kept from before is the back button between screens 1 and 2. The rating bar doesn't look good on top of screen 2. In addition the CSS of the rating bar (on top of screen 2) will be confusing to the user. If we leave it without the their decision highlighted they will think that they have to vote again and if we highlight their decision they will think they cannot change it. With the back button they can choose to go back and answer differently. 

Screen 1:
Screen 1

Screen 2:
Screen 2

Screen 3 - positive review:
Screen 3 - Positive Review

Screen 3 - negative review:
Screen 3 - Negative Review

File: octoprint_mrbeam/templates/review.jinja2 [hunk 1]
```diff
@@ -5,6 +5,7 @@
             <h3 id="review_question" class="review_title">{{ _('How likely are you to recommend Mr Beam to a friend or colleague?') }}</h3>
             <div id="rating_block">
                 <div class="rating">
+                    <span class="rating-label">Not at all likely</span>
```
Comment (andywerner) [inline octoprint_mrbeam/templates/review.jinja2]: pls mark for translation

File: octoprint_mrbeam/templates/review.jinja2 [hunk 2]
```diff
@@ -15,23 +16,47 @@
                     <button value="8">8</button>
                     <button value="9">9</button>
                     <button value="10">10</button>
+                    <span class="rating-label">Extremely likely</span>
```
Comment (andywerner) [inline octoprint_mrbeam/templates/review.jinja2]: pls mark for translation

File: octoprint_mrbeam/templates/review.jinja2 [hunk 3]
```diff
@@ -15,23 +16,47 @@
...
+            <div id="ask_user_details" class="review_hidden_part">
+                <label for="review_input_phone_email">{{ _('Phone or email: (optional)') }}
+                    <small>{{ _('If it\'s ok to contact you about your feedback.') }}</small>
+                </label>
+                <input type="text" class="form-control" id="review_input_phone_email">
+            </div>
+            <div id="positive_review" class="review_hidden_part">
+                <h3 class="review_title">{{ _('Thank you for your response.') }}</h3>
+                <p>{{ _('Positive reviews from awesome customers like you help others to feel confident about choosing Mr Beam too.') }}</p>
+                <p>{{ _('Could you take 60 seconds to go to the ') }}
```
Comment (andywerner) [inline octoprint_mrbeam/templates/review.jinja2]: Please bring this whole sentence in one translatable entity with placeholders for opening and closing tag. This is This is important for translations.

File: octoprint_mrbeam/templates/review.jinja2 [hunk 4]
```diff
@@ -15,23 +16,47 @@
...
+                    <a id="give_review_link" target="_blank" href="https://store.mr-beam.org/products/mr-beam-ii-dreamcut-air-filter-ii-bundle?utm_source=beamos&utm_medium=beamos&utm_campaign=review_screen#shopify-product-reviews">{{ _('Mr Beam Store') }}</a>
```
Comment (andywerner) [inline octoprint_mrbeam/templates/review.jinja2]: please use this URL: `https://mr-beam.org/beamos/review`

**Review by andywerner — CHANGES_REQUESTED** [review]
Here is a description on how to mark text in jinja files: https://sites.google.com/mr-beam.org/wiki/development/internationalization?authuser=0#h.p_lMBteIJOQst1

**Review by andywerner — APPROVED** [review]

**Pull request merged**